test/functional/aws_test.go

```diff
@@ -0,0 +1,162 @@
+	awsCredsSecretName         = "aws-creds"
+)
+
+var (
+	expectedResources = []string{
+		ThreeScaleResourceName,
+		UPSResourceName,
+		CRResourceName,
+		ClusterSSOResourceName,
+		ApplicationSSOResourceName,
+	}
+)
+
+func TestAWS(t *testing.T) {
+	//TODO
+}
+
+func testExpectedResourcesExist(t testing.T, f *framework.Framework, instances []*rds.DBInstance, cacheClusters []*elasticache.CacheCluster, buckets []*s3.Bucket, expectedResources []string) (map[string]bool, error) {
+	err := setAWSCredEnvar(f)
+	if err != nil {
+		fmt.Errorf("error :%w", err)
+		return nil, err
+	}
+	var checkedResources map[string]bool
+	//check RDS resources
+
+	for _, instance := range instances {
+		for _, name := range expectedResources {
+			if instance.DBName != aws.String(name) {
+				continue
+			}
+			if instance.MultiAZ != aws.Bool(true) {
+				t.Logf("instance: %v , has multi-AZ disabled, instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: Multi-AZ is disabled")
+			}
+			if instance.StorageEncrypted != aws.Bool(true) {
+				t.Logf("instance: %v , has database encryption disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: encryption is disabled")
+			}
+			if instance.PreferredBackupWindow == nil {
+				t.Logf("instance: %v , has automated backups disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: automated backups are disabled")
+			}
+			if instance.DeletionProtection != aws.Bool(true) {
+				t.Logf("instance: %v , has deletion protection disabled,  instance ARN: %v ", instance.DBName, instance.DBInstanceArn)
+				return nil, errors.New("error: deletion protection is disabled")
+			}
+			checkedResources[aws.StringValue(instance.DBName)] = true
+
+		}
+	}
+}
+
+func newAWSClients(session *session.Session) *awsClients {
+	return &awsClients{
+		rdsClient:         rds.New(session),
+		elasticacheClient: elasticache.New(session),
+		s3Client:          s3.New(session),
+	}
+}
+
+func setAWSCredEnvar(f *framework.Framework) error {
+	secret := &corev1.Secret{}
+	err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: awsCredsSecretName, Namespace: awsCredsNamespace}, secret)
+	if err != nil {
+		return fmt.Errorf("error getting secret: %v from cluster: %w ", awsCredsSecretName, err)
+	}
+	if string(secret.Data["aws_access_key_id"]) != "" && string(secret.Data["aws_secret_access_key"]) != "" {
+		os.Setenv("AWS_ACCESS_KEY_ID", string(secret.Data["aws_access_key_id"]))
+		os.Setenv("AWS_SECRET_ACCESS_KEY", string(secret.Data["aws_secret_access_key"]))
+		return nil
+	}
+	return fmt.Errorf("error setting %v envar, secret data may not exist", awsCredsSecretName)
+}
+
+// Wait for status of all resources to become "available", return true if all available
+func awsResourcesAvailable(dbinstances []*rds.DBInstance, cacheClusters []*elasticache.CacheCluster) bool {
```

Comment:
Is this function needed? We probably have other tests (e.g. the alerts test) that will pick up on whether or not the resources are available, so we shouldn't need to wait for that (because that might also never happen)

Reply:
I wasn't entirely sure, so just created this just in case, but I've no objection if this is to be deleted